shell/platform/darwin/ios/ios_switchable_gl_context.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_DARWIN_IOS_IOS_SWITCHABLE_GL_CONTEXT_H_
+#define FLUTTER_SHELL_PLATFORM_DARWIN_IOS_IOS_SWITCHABLE_GL_CONTEXT_H_
+
+#include "flutter/fml/macros.h"
+#include "flutter/fml/memory/thread_checker.h"
+#include "flutter/shell/common/gl_context_switch.h"
+
+@class EAGLContext;
+
+namespace flutter {
+
+struct DebugThreadChecker {
+  FML_DECLARE_THREAD_CHECKER(checker);
+};
+
+class IOSSwitchableGLContext final : public SwitchableGLContext {
+ public:
+  IOSSwitchableGLContext(EAGLContext& context);
```

Comment:
You should avoid passing non-const references, this will soon be a linter error:
https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs

It's especially egregious for NSObjects.  This class should retain and release the eaglcontext, too.

Reply:
This should actually be const. I'll update. This class doesn't retain the EAGLContext, all the EAGLContexts are retained by IOSRenderTarget and IOSContextGL. This class should just use the context and never out live either of the 2.